### PR TITLE
(chore) Dependency upgrades and improvements to linting and types

### PR DESCRIPTION
**Pull request by denniskigen**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [x] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [x] My work includes tests or is validated by existing tests.

## Summary

I’ve made various upgrades to dependencies, lint tooling and types in this PR. Specifically, these changes include:

- Bumps various packages to their latest versions.
- Updates the ESLint configuration to include the [recommended-type-checked](https://typescript-eslint.io/linting/configs/#projects-with-type-checking) plugin configuration for improved type safety. I’ve also turned off a few potentially conflicting rules.
- Replacing `pretty-quick` with `lint-staged` for managing pre-commit hooks. `pretty-quick` isn’t compatible with the latest version of Prettier.
- Improved type annotations.
- Adds an error notification to the Edit Question modal that gets shown when the name of the linked concept for a question cannot be resolved.

## Other

Changes to linting and formatting were inspired by https://www.joshuakgoldberg.com/blog/configuring-eslint-prettier-and-typescript-together/.

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+	"bracketSpacing": true,
+	"printWidth": 80,
```
Comment (ibacher) [inline .prettierrc:3]: Don't we usually use 120?

Comment (denniskigen) [reply .prettierrc:3]: [Yeah](https://github.com/search?q=org%3Aopenmrs%20printWidth&type=code). I'll switch back to that.

File: .eslintignore [hunk 2]
```diff
@@ -1,2 +1,3 @@
+e2e/**
```
[the remark this answers is not in the document]
Comment (denniskigen) [reply .eslintignore:1]: Yes, they should. I forgot to restore this.

**Review by ibacher — DISMISSED** [review]

**Comment by denniskigen** [comment]
Dunno why the e2e tests are failing @ibacher. I've posted a thread on the qa-support channel on Slack.

File: e2e/specs/forms-dashboard.spec.ts [hunk 3]
```diff
@@ -16,54 +11,53 @@ test.beforeEach(async ({ api }) => {
...
   });
 
   // Test the filter functionality
-  await test.step("Then I click the publish filter dropdown", async () => {
+  await test.step('Then I click the publish filter dropdown', async () => {
     await page
-      .getByRole("combobox", {
-        name: "Filter by publish status: All Open menu",
+      .getByRole('combobox', {
+        name: 'Filter by publish status: All Open menu',
```
Comment (ibacher) [inline e2e/specs/forms-dashboard.spec.ts:25]: I don't think this test could've ever succeeded. The DOM element with the role `combobox` is the button which only has the text "All".

File: e2e/specs/unpublish-form.spec.ts [hunk 4]
```diff
@@ -16,28 +11,28 @@ test.beforeEach(async ({ api }) => {
...
 
-  await test.step("And I click on a form I need to unpublish", async () => {
+  await test.step('And I click on a form I need to unpublish', async () => {
     await page.getByRole('row', { name: form.name }).getByLabel('Edit Schema').click();
   });
 
-  await test.step("Then I click on the unpublish form button and confirms the unpublication", async () => {
+  await test.step('Then I click on the unpublish form button and confirms the unpublication', async () => {
     await formBuilderPage.unpublishFormButton().click();
     await formBuilderPage.unpublishFormConfirmationButton().click();
```
Comment (ibacher) [inline e2e/specs/unpublish-form.spec.ts:31]: Watching the video, this doesn't seem to work. There's a failure "Cannot read properties of undefined (reading 'length')". This seems to trigger at the first `unpublishFormButton().click()`.

File: e2e/specs/forms-dashboard.spec.ts (hunk 3 above)
Comment (denniskigen) [reply e2e/specs/forms-dashboard.spec.ts:25]: And it works!

**Comment by denniskigen** [comment]
All fixed, @ibacher.

**Review by vasharma05 — DISMISSED** [review]
Thanks!

**Review by donaldkibet — APPROVED** [review]

**Review by vasharma05 — APPROVED** [review]

**Pull request merged**